### ISSUE TITLE
Rearrange Burger Menu Items [Dev-16.0]

### ISSUE THREAD

**Issue by hstaatsxx** [ConstructiveNews/squirrel-news-app#42]
User story:
**As a** User
**I want** the burger menu items arranged in a meaningful order
**So that** i can find my way around the menu better

Acceptance criteria:
_Given_ the burger menu icon is pressed
_When_ the menu expands
_Then_ the items are arranged in the given order

_Given_ the burger menu is opened
_When_ the "About" menu item is pressed
_Then_ the list of sub-items is expanded

> 1. Home
> 2. All issues
> 3. Favourite
> 4. Podcast
> 5. Donation
> 6. About
> 6.1 Team
> 7. Contact us
> 8. Settings
> 9. Imprint
> 10. Privacy policy

Spende evtl. anderes icon
Favoriten evtl. mit ionic bookmark icon, dann auch auf Teaser übernehmen (nochmal überdenken)
Impressum mit Paragraph icon
 
Dependencies:
none

DoD Checklist:
- [x] Acceptance criteria are met
- [x] Test by developer has been done
- [x] Code is commented if required
- [ ] Code/PR review has been done
- [x] Feature is documented
- [ ] Internal test by a 2nd Dev or the PO has been done

**Comment by hstaatsxx** [comment]
@olierxleben  @constructive-news 
Hi, I need your input on the decision on the icons.

1) Donation

- [ ] Heart inside of a circle
- [ ] Outlined gift icon


2) Favorite/Bookmark 

- [ ] previous select -> Heart outlined icon
- [ ] Bookmarks outlined

3. Imprint
Ionic or any react icon has "Paragraph" Icon. So I used other svg file to create it.
Since I use another svg file for it, it looks slightly different/thicker so users may notice.

Some images for the references:
<img width="376" alt="Bildschirmfoto 2022-12-12 um 19 30 04" src="https://user-images.githubusercontent.com/31039939/207126682-7b1dd366-703d-4085-bc60-6a0cd436ccbd.png">
<img width="379" alt="Bildschirmfoto 2022-12-12 um 19 28 16" src="https://user-images.githubusercontent.com/31039939/207126844-3c1cd655-6f5d-48a4-b9cf-fdb59fbec76b.png">

**Comment by hstaatsxx** [comment]
<img width="376" alt="Bildschirmfoto 2022-12-12 um 19 30 04" src="https://user-images.githubusercontent.com/31039939/207134426-689b9aaf-d454-4d0b-bcd0-8a30c7ece2ac.png">

I added a sub-menu to 'about'.  When a user clicks on 'Team' it will open the current 'about' page.
Is this okay?

**Comment by constructive-news** [comment]
The team submenu item + icon looks nice! We have to add a page now with the team content, just like here: https://squirrel-news.net/team/ where the submenu item should link to. Linking to the website might be easiest for now, as the team site is often updated (so we have to update it only once at the moment)

**Comment by constructive-news** [comment]
Thank you also for the icons! Is there a reason why you chose a heart with a circle and not just a heart?

And yes, the paragraph icon looks a bit different. Maybe we have to chose another one here then, after all.

I will look at all this with our UX designer Oi as soon as possible and will get back to you then. Just wanna avoid to much back and forth.

**Comment by hstaatsxx** [comment]
Thank you @constructive-news for the feedback.
I would not set the current (/about) path on the parent menu, cuz it will cause double onClick action.
1. Open collapsed menu
2. Open about page

How about add second submenu like
**About** 
**- Squirrel News** (/about link)
**- Team** (https://squirrel-news.net/team/)

I chose circled heart because we are still not sure about Favorite icon.

Please give me your final choices for following menu icons :)
- Favorite
- Donation
- Imprint

**Comment by constructive-news** [comment]
Hi again, 

I like the new imprint icon; we can keep it (see attachment); and we can also keep the bookmark icon for the Favourites (with ou and s). For the donation we can then take a normal heart (but the wording should not be "Donation", but "Support us").

[](u

![Image](https://user-images.githubusercontent.com/58957777/208750831-f3469193-bcf4-4b48-b50b-386eca8ecf72.png)
![Image](https://user-images.githubusercontent.com/58957777/208750830-2b3c62f1-5e01-4e9d-8e1b-96051315eeb5.png)

For the submenu we haven't talked about it yet. But what you write makes sense. We could just change the wording from "Squirrel News" into "About us" (with the same icon as "About"); change "Our Team" into "Team", and add "Contact" below, which isn't there yet (icon: <ion-icon name="mail-outline"></ion-icon>)

I think that should work for now

**Comment by constructive-news** [comment]
mail icon code was deleted. adding a screenshot 

![Image](https://user-images.githubusercontent.com/58957777/208754575-a8add235-c113-466b-9d08-2b760b88e31a.png)

**Comment by hstaatsxx** [comment]
Hi @constructive-news ,
thank you for your quick response.

- New Imprint icon & Favourites icon with bookmark can stay
- Wording will be updated in #49 ticket and after merging ticket in main in this issue as an update also
- I need a link for the 'Contact'! :) what should it be?

Thank you!
Regards

**Comment by constructive-news** [comment]
Let’s just take the Contact page on the website (the one with the contact form). In the long run we could also add a contact form right there in the app

**Comment by hstaatsxx** [comment]
Here is the documentation for the feature (as discussed in #46).

### Documentation
**Short description of the change**
In this issue, we updated our existing burger menu. 
Order or burger menu is changed and the 'Contact us' menu and sub-menus are added to the menu 'About'.
Icons are updated.

- Donation (Support Us) -> Outlined heart
- Imprint -> Outlined Receipt
- About -> Outlined circled information
- About Us -> same as About
- Favourite -> bookmarks

Current changes with text-variable #49 merged and updated with submenu items.

**Codebase Changes**
Another interface SubPage is added to SideMenu.tx which is used as an optional array field in the Page interface.
externalLink was removed in another issue but added back for the contact page connection.

Conditional rendering for menu items is added to implement optional submenus.
Submenu item rendering is also added. All code is written for generic use. Adding further submenus should be easily done,

**Visual Documentation**
Visual documentation for rearranged burger menu:

https://user-images.githubusercontent.com/31039939/208971637-fb415fcb-8b8f-42f0-a31b-8158a71629fa.mov
